public/components/client-projects/client-projects.js

```diff
@@ -65,17 +64,16 @@ export const ClientProjectVM = DefineMap.extend({
         this.newClientName = "";
         this.toggleClientInput();
         this.selectedClientId = "__new__";
-
       });
-    }
-    else {
-      promise = this.projects.then(projects => {
-        projects.forEach(project => {
-          if( project._id === selectedClientId ) {
+    } else {
+      promise = this.projects.then((projects) => {
+        const self = this;
```

Comment:
doesn't this make it more confusing since the `this` inside the arrow function is actually the outer scope?